js/components/skipControls.js

```diff
@@ -0,0 +1,276 @@
+      <HoldControlButton
+        {...this.props}
+        className="oo-center-button oo-skip-forward"
+        icon="replay"
+        focusId={CONSTANTS.SKIP_CTRLS_KEYS.SKIP_FORWARD}
+        ariaLabel={skipForwardAriaLabel}
+        onClick={this.onSkipForward}>
+        <span className="oo-btn-counter">{skipTimes.forward}</span>
+      </HoldControlButton>
+    );
+
+    buttonTemplate[CONSTANTS.SKIP_CTRLS_KEYS.NEXT_VIDEO] = (
+      <ControlButton
+        {...this.props}
+        className="oo-next-video"
+        icon="nextVideo"
+        focusId={CONSTANTS.SKIP_CTRLS_KEYS.NEXT_VIDEO}
+        ariaLabel={CONSTANTS.ARIA_LABELS.NEXT_VIDEO}
+        disabled={!this.props.config.hasNextVideos}
+        onClick={this.onNextVideo}>
+      </ControlButton>
+    );
+    return buttonTemplate;
+  },
+
+  /**
+   * Determines whether or not the button with the particular id can be displayed
+   * considering the current player state and configuration.
+   * @private
+   * @param {string} buttonId The id of the button we want to check
+   * @return {boolean} True if the button should be displayed, false otherwise
+   */
+  shouldDisplayButton: function(buttonId) {
+    var config = this.props.config;
+    var isSingleVideo = !config.hasPreviousVideos && !config.hasNextVideos;
+    var duration = Utils.getPropertyValue(this.props.controller, 'state.duration');
+
+    var isPrevNextButton = (
+      buttonId === CONSTANTS.SKIP_CTRLS_KEYS.PREVIOUS_VIDEO ||
+      buttonId === CONSTANTS.SKIP_CTRLS_KEYS.NEXT_VIDEO
+    );
+    var isSkipButton = (
+      buttonId === CONSTANTS.SKIP_CTRLS_KEYS.SKIP_BACKWARD ||
+      buttonId === CONSTANTS.SKIP_CTRLS_KEYS.SKIP_FORWARD
+    );
+
+    var isDisabled = (
+      (isSkipButton && !duration) ||
+      (isPrevNextButton && isSingleVideo)
+    );
+    return !isDisabled;
+  },
+
+  /**
+   * Parses the skin.json's skip button configuration and returns the ids of the
+   * enabled buttons sorted by index in ascending order.
+   * @private
+   * @return {array} An array of button objects. Each object contains the id and index
+   * of the button.
+   */
+  getSortedButtonEntries: function() {
+    var buttons = [];
+    var buttonConfig = Utils.getPropertyValue(
+      this.props.skinConfig,
+      'skipControls.buttons',
+      {}
+    );
+    // Find de ids and indexes of all enabled buttons
+    for (var buttonId in buttonConfig) {
+      var button = buttonConfig[buttonId];
+
+      if (button && button.enabled && this.shouldDisplayButton(buttonId)) {
+        buttons.push({
+          id: buttonId,
+          index: button.index
+        });
+      }
+    }
+    // Sort by index in ascending order
+    buttons.sort(function(a, b) {
+      return a.index - b.index;
+    });
+    return buttons;
+  },
+
+  render: function() {
+    var buttons = this.getSortedButtonEntries();
+    // Nothing to render if we don't have buttons
+    if (!buttons.length) {
```

Comment:
What happens from an end user perspective when this happens? Will they just be sitting on a blank screen? Is this something we want to consider avoiding?